app/lib/core/firstDay.dart

```diff
@@ -0,0 +1,59 @@
+/// Enum representing the first day of the week
+enum TraleFirstDay {
+  /// Sunday as the first day of the week
+  sunday,
+
+  /// Monday as the first day of the week
+  monday,
+
+  /// Saturday as the first day of the week
+  saturday,
+}
+
+/// Extension for additional properties and methods for TraleFirstDay
+extension TraleFirstDayExtension on TraleFirstDay {
+  /// Mapping of TraleFirstDay values to DateTime weekday values
+  static const _weekdayMapping = <TraleFirstDay, int>{
+    TraleFirstDay.sunday: 0,
+    TraleFirstDay.monday: 1,
+    TraleFirstDay.saturday: 6,
+  };
+
+  /// Mapping of TraleFirstDay values to their localized names
+  static const _localizedNames = <TraleFirstDay, String>{
+    TraleFirstDay.sunday: 'Sunday',
+    TraleFirstDay.monday: 'Monday',
+    TraleFirstDay.saturday: 'Saturday',
```

Comment:
The names of the weekdays can be obtained from the intl package. I guess by doing so you should get a list with all weekdays.
```dart
  List<String> weekdayStrings(BuildContext context) {  
    final Locale activeLocale = Localizations.localeOf(context);
    return DateFormat.EEEE(activeLocal).dateSymbols.STANDALONEWEEKDAYS;
  }
```